cluster-operator/pom.xml

```diff
@@ -334,6 +334,11 @@
             <resource>
                 <directory>src/main/resources</directory>
             </resource>
+            <resource>
+                <!-- separate folder for filtered resources so that non-filtered files don't have their ${} replaced -->
+                <directory>src/main/resources-filtered</directory>
+                <filtering>true</filtering>
+            </resource>
```

Comment:
So I originally used this, but for local development `getPackage().getImplementationVersion()` always returns null for unit tests, I read into this and it appears that unless you're running the actual JARs, the information you need for this info is not actually populated, hence why it appears in the main log lines.

Now I can just set the version like this, but then this means from a unit test point of view, unless we do some clever mocking for setting the versions field, or falling back to a default if its null, it will always be setting the key as null which will make for some confusing tests.
Using these templated resources gets around this as the resources will be templated prior to the unit test phase.

If you have any advice/thoughts on how to get `Main.class.getPackage().getImplementationVersion()` to work at unit test level I would absolutely implement the version as this, but I am unsure whether it is wise to do this given it will always be null for unit tests.

Reply:
Yeah, I guess you are right. That call gets it from the JAR itself. That seems unfortunate :-/.